marimo/_ast/sql_visitor.py

```diff
@@ -315,109 +313,52 @@ def find_sql_refs(
     Returns:
         A list of table and schema names referenced in the statement.
     """
-    if not DependencyManager.duckdb.has():
+
+    # Use sqlglot to parse ast (https://github.com/tobymao/sqlglot/blob/main/posts/ast_primer.md)
+    if not DependencyManager.sqlglot.has():
         return []
 
-    import duckdb
+    from sqlglot import exp, parse
+    from sqlglot.optimizer.scope import build_scope
 
-    tokens = duckdb.tokenize(sql_statement)
-    token_extractor = TokenExtractor(
-        sql_statement=sql_statement, tokens=tokens
-    )
     refs: list[str] = []
-    cte_names: set[str] = set()
-    i = 0
 
-    # First pass - collect CTE names
-    while i < len(tokens):
-        if token_extractor.is_keyword(i, "with"):
-            i += 1
-            # Handle optional parenthesis after WITH
-            if token_extractor.token_str(i) == "(":
-                i += 1
-            while i < len(tokens):
-                if token_extractor.is_keyword(i, "select"):
-                    break
-                if (
-                    token_extractor.token_str(i) == ","
-                    or token_extractor.token_str(i) == "("
-                ):
-                    i += 1
-                    continue
-                cte_name = token_extractor.strip_quotes(
-                    token_extractor.token_str(i)
+    def append_refs_from_table(table: exp.Table) -> None:
+        if table.catalog == "memory":
+            # Default in-memory catalog, only include table name
+            refs.append(table.name)
+        else:
+            # We skip schema if there is a catalog
+            # Because it may be called "public" or "main" across all catalogs
+            # and they aren't referenced in the code
+            if table.catalog:
+                refs.append(table.catalog)
+            elif table.db:
+                refs.append(table.db)  # schema
+
+            if table.name:
+                refs.append(table.name)
```

Comment:
For `my_catalog.my_schema.my_table`, where `my_catalog` is explicitly given in the code, I wonder if the ref should just be `my_catalog` instead of `my_catalog` and `my_table`? Because two separate catalogs could have overlapping table names. Analogous to how in Python, the ref for `my_instance.my_variable` is just `my_instance`. 

A genuine question. Curious what makes more sense to you.

Reply:
I think right now, `my_catalog.my_schema.my_table` generates a def of just `my_table` -- so the logic of adding `my_table` as a ref makes sense in the context of this PR. Perhaps we can revisit this question later.

Reply:
Yea this was a known limitation / edge case when we added this, but it was the lesser of two evils. If using sqlglot always gives us a valid catalog, we could only use fully qualified/verbose table names. 

But given this does the same logic as before, I think we can look into it later (agreed with @akshayka)

Reply:
can revisit this when we modify `find_sql_defs` too